### ISSUE TITLE
Is it possible to use SwinTransformer instead of ResNet?

### ISSUE THREAD

**Issue by omniaalwazzan** [mahmoodlab/CLAM#117]
Hi

I realise my question is a bit broad, but I was curious whether I could use SwinTransformer as a backbone network instead of ResNet.

SwinTransofrmer seemed to gather far more information from an input image than ResNet when I read the study (since it focuses on a 4x4 image patch and gradually increases). In contrast to the WSI, where we have a slide level label, I found that the Swin transformer for a segmentation task worked best with images that include a class label. 

Do you think it is worth trying Swin Transformer with CLAM or it's not applicable?

Thanks in advance.

**Comment by micdic** [comment]
Hi omniaalwazzan,
i've tried to replace the Resnet50 with a small Resnet18  (paper here https://arxiv.org/pdf/2011.13971.pdf), in order to test features extraction with a pretrained network on CPTAC (not ImageNet). The network is smaller with only 512 features output vector size, and initially my results were worst on a cancer subtyping problem. Still working oi it...
But you are referring to feature extraction or segmentation?

**Comment by omniaalwazzan** [comment]
> Hi omniaalwazzan, i've tried to replace the Resnet50 with a small Resnet18 (paper here https://arxiv.org/pdf/2011.13971.pdf), in order to test features extraction with a pretrained network on CPTAC (not ImageNet). The network is smaller with only 512 features output vector size, and initially my results were worst on a cancer subtyping problem. Still working oi it... But you are referring to feature extraction or segmentation?

Hi micdic, 

Thanks for sharing your knowledge. Interesting, I am wondering how big is your dataset?  I'm assuming your dataset has fewer than 500 wsi.



I'm talking about feature extraction; I know Swin-transformer is used for segmentation, but they also have pretrained weights; however, I don't think it's appropriate to use it in the CLAM pipeline instead of ResNet because CALM patches are arrays of coords and features in h5 format, whereas Swin- transformer's architecture is different. Confused, That's why I inquired here:)

If I use the Swin-transformer for segmentation, I doubt I'll be able to use CLAM because the resulting feature vector will have a different diminetion of the clam.

I hope this has answered your question:)

**Comment by micdic** [comment]
Hi omniaalwazzan,
i have a dataset with 250 WSI per class (more or less).
Btw, i've adapted easily the features vector size (output from the features extraction in the inner patch classifier of CLAM) to the pipeline.  Just a couple of code updates in the model, replacing input size 1024 with 512. Then the features will be stored in h5 format after...

**Comment by omniaalwazzan** [comment]
Thanks ,I see your point. I'll take your advice regarding the feature size into account and will share any improved results with you if you're interested.

Your dataset is comparable in size to mine, however, I didn't consider changing the ResNet architecture because I obtained about the same AUC as the CLAM working on 75% training data. Thus, I'd like to experiment with a backbone network other than ResNet to observe how the model acts!

To be honest, I am quite interested in the transformer, which is why I am seeking such a solution.
